open-craft/edx-analytics-pipeline
# SE-1448 Use bearer token type for calls to Course List API

Changes to the JWT authentication for edxapp result in errors when invalid JWT authentication credentials are provided, so bearer tokens must be used instead.

**Testing instructions**

* See the previous build failures for ProblemResponseReport, e.g. [build#7](http://127.0.0.1:8080/job/ProblemResponseReport/7/console): 

<details><summary>HTTPError: 500 Server Error: Internal Server Error for url: https://www.epodx.org/api/courses/v1/courses/?page_size=100</summary>

```
2019-08-21 02:46:24,714 ERROR 14842 [luigi-interface] worker.py:213 - [pid 14842] Worker Worker(salt=869664237, workers=1, host=ip-172-31-59-16, username=hadoop, pid=14842, sudo_user=hadoop) failed    PullCourseListApiData(warehouse_path=s3://epodx-edxanalytics/warehouse/hive/, datetime=2019-08-21T023013, partition_format=%Y-%m-%d, api_root_url=https://www.epodx.org/api/courses/v1/courses/)
Traceback (most recent call last):
 File "/mnt/var/lib/analytics-tasks/automation/venv/src/luigi/luigi/worker.py", line 194, in run
   new_deps = self._run_get_new_deps()
 File "/mnt/var/lib/analytics-tasks/automation/venv/src/luigi/luigi/worker.py", line 131, in _run_get_new_deps
   task_gen = self.task.run()
 File "/var/lib/analytics-tasks/automation/venv/local/lib/python2.7/site-packages/edx/analytics/tasks/insights/course_list.py", line 114, in run
   for response in client.paginated_get(self.api_root_url, params=params, pagination_key=_pagination):
 File "/var/lib/analytics-tasks/automation/venv/local/lib/python2.7/site-packages/edx/analytics/tasks/util/edx_api_client.py", line 222, in paginated_get
   response, next_url = get_resource_with_retry(next_url)
 File "/var/lib/analytics-tasks/automation/venv/local/lib/python2.7/site-packages/edx/analytics/tasks/util/retry.py", line 56, in retry_function
   return func(*args, **kwargs)
 File "/var/lib/analytics-tasks/automation/venv/local/lib/python2.7/site-packages/edx/analytics/tasks/util/edx_api_client.py", line 213, in get_resource_with_retry
   raw_response.raise_for_status()
 File "/var/lib/analytics-tasks/automation/venv/local/lib/python2.7/site-packages/requests/models.py", line 935, in raise_for_status
   raise HTTPError(http_error_msg, response=self)
HTTPError: 500 Server Error: Internal Server Error for url: https://www.epodx.org/api/courses/v1/courses/?page_size=100
```
</details>

* Note that running this branch (check `TASKS_BRANCH`) has fixed the authentication issue, e.g. [build#8](http://127.0.0.1:8080/job/ProblemResponseReport/8/consoleFull). 

**Author Notes & Concerns**

* This change will need to be submitted upstream.

**Reviewer**
